c_binding/include/libstoragemgmt/libstoragemgmt.h

```diff
@@ -1029,6 +1030,102 @@ int LSM_DLL_EXPORT lsm_volume_ident_led_set(lsm_connect *c,
+ * New in version 1.3
+ * @param [in]      conn            Valid connection @see
+ *                                  lsm_connect_password
+ * @param[in]       search_key      Search key (NULL for all)
+ * @param[in]       search_value    Search value
+ * @param [out]     batteries       An array of lsm_battery types
+ * @param [out]     count           Number of batteries
+ * @param [in]      flags           Reserved set to zero
+ * @return LSM_ERR_OK on success else error reason
+ */
+int LSM_DLL_EXPORT lsm_battery_list(lsm_connect *conn,
+                                    const char *search_key,
+                                    const char *search_value,
+                                    lsm_battery ***batteries, uint32_t *count,
+                                    lsm_flag flags);
+
+/**
+ * Query RAM cache information for the desired volume.
+ * New in version 1.3.
+ * @param[in] c             Valid connection
+ * @param[in] volume        A single lsm_volume
+ * @param[in] flags         Reserved, set to 0
+ * @param[out] write_cache_policy
+ *                         The write cache policy. Valid values are:
+ *                          * LSM_VOLUME_WRITE_CACHE_POLICY_WRITE_BACK
+ *                            The storage system will use write back mode if
+ *                            cache hardware found. In write back mode, I/O
+ *                            write data is done asynchronously to the cache and
+ *                            to the backing store(normally disk), the
+ *                            acknowledgement will send back to I/O initiator
+ *                            once cache stored all data.
+ *                          * LSM_VOLUME_WRITE_CACHE_POLICY_AUTO
+ *                            The controller will use write back mode when
+ *                            battery/capacitor is in good health, otherwise,
+ *                            write through mode.
+ *                          * LSM_VOLUME_WRITE_CACHE_POLICY_WRITE_THROUGH
+ *                              The storage system will use write through mode.
+ *                          * LSM_VOLUME_WRITE_CACHE_POLICY_UNKNOWN
+ *                              Plugin failed to detect this setting.
+ * @param[out] write_cache_status
+ *                         The status of write cache. Valid values are:
+ *                          * LSM_VOLUME_WRITE_CACHE_STATUS_WRITE_THROUGH
+ *                          * LSM_VOLUME_WRITE_CACHE_STATUS_WRITE_BACK
+ *                          * LSM_VOLUME_WRITE_CACHE_STATUS_UNKNOWN
+ * @param[out] read_cache_policy
+ *                          The policy for read cache. Valid values are:
+ *                          * LSM_VOLUME_READ_CACHE_POLICY_ENABLED
+ *                              Read cache is enabled, when reading I/O on
+ *                              previous unchanged written I/O or read I/O in
+ *                              cache will be return I/O initiator immediately
```

Comment:
will be returned to the I/O initiator immediately without checking the backing store